### PR TITLE
cmd/openshift-install: switch bootstrap-complete watcher to polling

### PR DESCRIPTION
**Pull request by sttts**
Get rid of broken watcher code. Polling is by far enough here.

https://github.com/openshift/installer/pull/1272#issuecomment-465946993 sums up the issues.

### CODE REVIEW

File: cmd/openshift-install/create.go [hunk 1]
```diff
@@ -290,48 +289,13 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
...
-				return false, nil
-			}
-
-			logrus.Debugf("added %s: %s", event.Name, event.Message)
-			return event.Name == "bootstrap-complete", nil
-		},
-	)
-	if err != nil {
+	if err := wait.PollImmediate(time.Second, eventTimeout, func() (done bool, err error) {
+		if _, err := events.Get("bootstrap-complete", metav1.GetOptions{}); err == nil {
```
Comment (mfojtik) [inline cmd/openshift-install/create.go]: do we want to check for IsNotFound(err) here and return any other errors?

File: cmd/openshift-install/create.go [hunk 2]
```diff
@@ -290,48 +289,13 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
...
-			if watchEvent.Type != watch.Added {
-				return false, nil
-			}
-
-			logrus.Debugf("added %s: %s", event.Name, event.Message)
-			return event.Name == "bootstrap-complete", nil
-		},
-	)
-	if err != nil {
+	if err := wait.PollImmediate(time.Second, eventTimeout, func() (done bool, err error) {
```
Comment (tnozicka) [inline cmd/openshift-install/create.go]: this should be using https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/watch/until.go#L110

**Comment by crawford** [comment]
This needs an explanation in the commit message. You cannot make claims like "Get rid of broken watcher code" without providing some background.

**Comment by tnozicka** [comment]
`Until` was never meant for this case as its own comments state it is meant for tests where you care about seeing edges and you'd rather fail then miss them. By design it can never recover from RV too old error. You should have been using `UntilWithSync`, informer based with re-list (for level based use cases). I'd have explained it you if you had asked before taking code from live upstream PR and making modifications to it. Also taking the wrong function.

In addition to taking a function not fit for the use case @sttts has been seeing several cases when the Until hot loops on restarts. So this is broken by design and the implementation seems to have bugs as well.

We have made it clear to @wking on slack 27 days ago in a hope to fix it. I know @sttts has been encountering failures since, I am not sure about the frequency, but he can talk to that.

File: cmd/openshift-install/create.go (hunk 1 above)
Comment (sttts) [reply cmd/openshift-install/create.go]: Errors at this stage are normal. I would only output them and keep trying.

**Comment by sttts** [comment]
Part of https://github.com/openshift/installer/pull/1295 now.

**Pull request closed without merging**